internal/server/data/access_key.go

```diff
@@ -113,21 +109,22 @@ func ListAccessKeys(db GormTxn, p *models.Pagination, selectors ...SelectorFunc)
 	return list[models.AccessKey](db, p, selectors...)
 }
 
-func GetAccessKey(tx GormTxn, selectors ...SelectorFunc) (*models.AccessKey, error) {
-	db := tx.GormDB()
-	// GetAccessKey by keyID needs to not set an organization_id in the query.
-	// keyID should be globally unique.
-	for _, selector := range selectors {
-		db = selector(db)
-	}
-	result := new(models.AccessKey)
-	if err := db.Model(result).First(result).Error; err != nil {
-		if errors.Is(err, gorm.ErrRecordNotFound) {
-			return nil, internal.ErrNotFound
-		}
-		return nil, err
+// GetAccessKey using the keyID. Note that the keyID is globally unique, so
+// this query is not scoped by an organization_id.
+func GetAccessKey(tx ReadTxn, keyID string) (*models.AccessKey, error) {
```

Comment:
In production code we only ever queried by `keyID`, which I think is expected. In tests we can also just as easily query by this ID, so I changed this function to only accept this one argument.

In the future we could also make this an options struct, but I think it would be good to keep everything using keyID if possible.  We may also want to consider making the keyID the primary key and removing ID.